packages/material-ui-unstyled/src/utils/useResizeObserver.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { ResizeObserver as ResizeObserverPolyfill } from 'resize-observer';
```

Comment:
If imported, it needs to be a direct dependency of `@material-ui/unstyled`. But that's generally not what we want with a polyfill. App developers need to decide that. 

At the same time, we can't require `ResizeObserver` since that's not matching our browser support matrix.

The point about the abstraction still applies. Do not abstract before you have at least 3 usage sites. Trust me, that helps you identify the correct abstraction.

Reply:
Thanks. I created a mock resize observer so that there is no need to import or have a polyfill as a dependency. I also deleted the hook. https://github.com/mui-org/material-ui/pull/27791/commits/d591725a0a15d3e94b5969c305ac3d2c48c3439b

Reply:
The dummy ponyfill sounds similar to how we gracefully not support features with IE 11. Here, I assume that adding a note about the need to call the "update" function with a ref when the width changes on older browsers would be enough? A compromise.

Reply:
Are you referring to something like this?
```
function MyTabs() {
  const tabsActions = React.useRef();

  React.useEffect(() => {
    if (tabsActions.current) {
      tabsActions.current.updatePosition();
    }
  }, [x]);

  return <Tabs action={tabsActions}>{items}</Tabs>;
}
```

If so, should I add such note to docs? or just as comments in the component code?

Reply:
This is the current API we are recommending developers to use. I recall creating a demo for it a long time ago on some of the GitHub issues. We have lived with it for two years, I think that we can continue in the future without doing more effort, browser supports is improving over time. I'm resolving this thread.